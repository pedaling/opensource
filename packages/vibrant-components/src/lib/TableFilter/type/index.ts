export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
