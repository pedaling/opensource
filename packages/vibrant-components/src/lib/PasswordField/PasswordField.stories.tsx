import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { VStack } from '../VStack';
import { PasswordField } from './PasswordField';

export default {
  title: 'PasswordField',
  component: PasswordField,
  args: {
    label: '비밀번호',
    helperText: '비밀번호을 입력해주세요',
    size: 'lg',
  },
} as ComponentMeta<typeof PasswordField>;

export const Basic: ComponentStory<typeof PasswordField> = props => (
  <VStack width="100%" p={20}>
    <PasswordField {...props} />
  </VStack>
);
