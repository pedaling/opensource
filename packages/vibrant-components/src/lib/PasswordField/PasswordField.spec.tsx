import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { PasswordField } from './PasswordField';

describe('<PasswordField />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  describe('when size is lg', () => {
    beforeEach(async () => {
      renderer = render(<PasswordField size="lg" label="label" />);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when size is md', () => {
    beforeEach(async () => {
      renderer = render(<PasswordField size="md" label="label" />);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when size is sm', () => {
    beforeEach(async () => {
      renderer = render(<PasswordField size="sm" label="label" />);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
