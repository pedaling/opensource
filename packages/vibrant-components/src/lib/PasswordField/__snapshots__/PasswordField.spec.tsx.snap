// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordField /> when size is lg match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0;
  padding-right: 15px;
  width: 100%;
  height: 50px;
  border-width: 1px;
  border-style: solid;
  border-color: #00000026;
  background-color: #ffffff;
  border-radius: 2px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
  cursor: text;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: text;
}

.emotion-5 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  position: absolute;
  z-index: 1;
  left: 15px;
  right: 12px;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-7 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-top: 23px;
  padding-left: 15px;
  padding-bottom: 7px;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #00000000;
  border-width: 0;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.875rem;
  line-height: 1.125rem;
  width: 100%;
  padding: 0;
  height: 50px;
  padding-top: 23px;
  padding-left: 15px;
  padding-right: 12px;
  padding-bottom: 7px;
}

.emotion-8::-webkit-input-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-8::-moz-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-8:-ms-input-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-8::placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-9 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-top: 23px;
  padding-right: 12px;
  padding-bottom: 7px;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
  fill: #5b5b5b;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
          tabindex="0"
        >
          <span
            class="emotion-5"
            style="top: 15px; font-size: 0.875rem; line-height: 1.125rem;"
          >
            label
          </span>
        </div>
        <div
          class="emotion-6"
        >
          <span
            class="emotion-7"
          />
          <input
            autocomplete="current-password"
            class="emotion-8"
            inputmode="text"
            placeholder=""
            type="password"
            value=""
          />
          <span
            class="emotion-9"
          />
        </div>
      </div>
      <div
        class="emotion-2"
      >
        <div
          class="emotion-11"
          tabindex="0"
        >
          <svg
            class="emotion-12"
            data-testid="eyeoff-thin"
            viewBox="0 0 24 24"
          >
            <path
              class="emotion-13"
              d="M21.25 12.75c0 .15.1.25.25.25h1.25c.15 0 .25-.1.25-.25-.05-2.2-.85-4.25-2.1-5.9L19.65 8.1c.95 1.3 1.55 2.9 1.6 4.65Zm.7-9.8-.9-.9c-.1-.1-.25-.1-.35 0L18.35 4.4c-1.8-1.2-4-1.9-6.35-1.9-6 0-10.85 4.55-11 10.25 0 .15.1.25.25.25H2.5c.15 0 .25-.1.25-.25C2.9 8.05 7 4.25 12 4.25c1.85 0 3.6.55 5.05 1.45L13.9 8.85c-.55-.25-1.2-.35-1.9-.35-3.3 0-6 2.7-6 6 0 .7.1 1.35.35 1.95L2.1 20.7c-.1.1-.1.25 0 .35l.9.9c.1.1.25.1.35 0L21.9 3.3c.1-.1.15-.25.05-.35ZM7.8 15c-.05 0-.05 0 0 0 0-.15-.05-.35-.05-.5 0-2.35 1.9-4.25 4.25-4.25.15 0 .35 0 .5.05L7.8 15Zm8.35.4c-.35 1.6-1.65 2.9-3.25 3.25-1.2.25-2.35 0-3.25-.6L8.4 19.3c1.2.9 2.8 1.4 4.5 1.15 2.55-.4 4.65-2.45 5.05-5.05.25-1.7-.2-3.3-1.15-4.5l-1.25 1.25c.6.9.85 2.05.6 3.25Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PasswordField /> when size is md match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0;
  padding-right: 9px;
  width: 100%;
  height: 38px;
  border-width: 1px;
  border-style: solid;
  border-color: #00000026;
  background-color: #ffffff;
  border-radius: 2px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
  cursor: text;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: text;
}

.emotion-5 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  position: absolute;
  z-index: 1;
  left: 9px;
  right: 8px;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-7 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-top: 13px;
  padding-left: 9px;
  padding-bottom: 3px;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #00000000;
  border-width: 0;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.875rem;
  line-height: 1.125rem;
  width: 100%;
  padding: 0;
  height: 38px;
  padding-top: 13px;
  padding-left: 9px;
  padding-right: 8px;
  padding-bottom: 3px;
}

.emotion-8::-webkit-input-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-8::-moz-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-8:-ms-input-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-8::placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-9 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-top: 13px;
  padding-right: 8px;
  padding-bottom: 3px;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
  fill: #5b5b5b;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
          tabindex="0"
        >
          <span
            class="emotion-5"
            style="top: 9px; font-size: 0.875rem; line-height: 1.125rem;"
          >
            label
          </span>
        </div>
        <div
          class="emotion-6"
        >
          <span
            class="emotion-7"
          />
          <input
            autocomplete="current-password"
            class="emotion-8"
            inputmode="text"
            placeholder=""
            type="password"
            value=""
          />
          <span
            class="emotion-9"
          />
        </div>
      </div>
      <div
        class="emotion-2"
      >
        <div
          class="emotion-11"
          tabindex="0"
        >
          <svg
            class="emotion-12"
            data-testid="eyeoff-thin"
            viewBox="0 0 24 24"
          >
            <path
              class="emotion-13"
              d="M21.25 12.75c0 .15.1.25.25.25h1.25c.15 0 .25-.1.25-.25-.05-2.2-.85-4.25-2.1-5.9L19.65 8.1c.95 1.3 1.55 2.9 1.6 4.65Zm.7-9.8-.9-.9c-.1-.1-.25-.1-.35 0L18.35 4.4c-1.8-1.2-4-1.9-6.35-1.9-6 0-10.85 4.55-11 10.25 0 .15.1.25.25.25H2.5c.15 0 .25-.1.25-.25C2.9 8.05 7 4.25 12 4.25c1.85 0 3.6.55 5.05 1.45L13.9 8.85c-.55-.25-1.2-.35-1.9-.35-3.3 0-6 2.7-6 6 0 .7.1 1.35.35 1.95L2.1 20.7c-.1.1-.1.25 0 .35l.9.9c.1.1.25.1.35 0L21.9 3.3c.1-.1.15-.25.05-.35ZM7.8 15c-.05 0-.05 0 0 0 0-.15-.05-.35-.05-.5 0-2.35 1.9-4.25 4.25-4.25.15 0 .35 0 .5.05L7.8 15Zm8.35.4c-.35 1.6-1.65 2.9-3.25 3.25-1.2.25-2.35 0-3.25-.6L8.4 19.3c1.2.9 2.8 1.4 4.5 1.15 2.55-.4 4.65-2.45 5.05-5.05.25-1.7-.2-3.3-1.15-4.5l-1.25 1.25c.6.9.85 2.05.6 3.25Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PasswordField /> when size is sm match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0;
  padding-right: 7px;
  width: 100%;
  height: 30px;
  border-width: 1px;
  border-style: solid;
  border-color: #00000026;
  background-color: #ffffff;
  border-radius: 2px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
  cursor: text;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: text;
}

.emotion-5 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  position: absolute;
  z-index: 1;
  left: 7px;
  right: 6px;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-7 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-top: 12px;
  padding-left: 7px;
  padding-bottom: 0;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 400;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #00000000;
  border-width: 0;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.875rem;
  line-height: 1.125rem;
  width: 100%;
  padding: 0;
  height: 30px;
  padding-top: 12px;
  padding-left: 7px;
  padding-right: 6px;
  padding-bottom: 0;
}

.emotion-8::-webkit-input-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-8::-moz-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-8:-ms-input-placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-8::placeholder {
  color: #949494;
  -webkit-text-fill-color: #949494;
  text-fill-color: #949494;
}

.emotion-9 {
  text-align: left;
  color: #5b5b5b;
  -webkit-text-fill-color: #5b5b5b;
  text-fill-color: #5b5b5b;
  display: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-top: 12px;
  padding-right: 6px;
  padding-bottom: 0;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 400;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
  fill: #5b5b5b;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
          tabindex="0"
        >
          <span
            class="emotion-5"
            style="top: 7px; font-size: 0.75rem; line-height: 1rem;"
          >
            label
          </span>
        </div>
        <div
          class="emotion-6"
        >
          <span
            class="emotion-7"
          />
          <input
            autocomplete="current-password"
            class="emotion-8"
            inputmode="text"
            placeholder=""
            type="password"
            value=""
          />
          <span
            class="emotion-9"
          />
        </div>
      </div>
      <div
        class="emotion-2"
      >
        <div
          class="emotion-11"
          tabindex="0"
        >
          <svg
            class="emotion-12"
            data-testid="eyeoff-thin"
            viewBox="0 0 24 24"
          >
            <path
              class="emotion-13"
              d="M21.25 12.75c0 .15.1.25.25.25h1.25c.15 0 .25-.1.25-.25-.05-2.2-.85-4.25-2.1-5.9L19.65 8.1c.95 1.3 1.55 2.9 1.6 4.65Zm.7-9.8-.9-.9c-.1-.1-.25-.1-.35 0L18.35 4.4c-1.8-1.2-4-1.9-6.35-1.9-6 0-10.85 4.55-11 10.25 0 .15.1.25.25.25H2.5c.15 0 .25-.1.25-.25C2.9 8.05 7 4.25 12 4.25c1.85 0 3.6.55 5.05 1.45L13.9 8.85c-.55-.25-1.2-.35-1.9-.35-3.3 0-6 2.7-6 6 0 .7.1 1.35.35 1.95L2.1 20.7c-.1.1-.1.25 0 .35l.9.9c.1.1.25.1.35 0L21.9 3.3c.1-.1.15-.25.05-.35ZM7.8 15c-.05 0-.05 0 0 0 0-.15-.05-.35-.05-.5 0-2.35 1.9-4.25 4.25-4.25.15 0 .35 0 .5.05L7.8 15Zm8.35.4c-.35 1.6-1.65 2.9-3.25 3.25-1.2.25-2.35 0-3.25-.6L8.4 19.3c1.2.9 2.8 1.4 4.5 1.15 2.55-.4 4.65-2.45 5.05-5.05.25-1.7-.2-3.3-1.15-4.5l-1.25 1.25c.6.9.85 2.05.6 3.25Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
